src/oc_erchef/apps/oc_chef_wm/src/chef_wm_actions_queue_monitoring.erl

```diff
@@ -0,0 +1,472 @@
+%% CALLS
+handle_call(is_queue_at_capacity, _From, #state{queue_at_capacity =
+                                                QueueAtCapacity} = State) ->
+    {reply, QueueAtCapacity, State};
+
+handle_call({override_queue_at_capacity, AtCapacity}, _From, State) ->
+    lager:info("Manually setting Queue Monitor queue_at_capacity ~p", [AtCapacity]),
+    {reply, ok, State#state{queue_at_capacity = AtCapacity}};
+handle_call(status, _From, State) ->
+    % return a term to be converted to JSON
+    Stats = [{queue_at_capacity,State#state.queue_at_capacity},
+             {dropped_since_last_check, State#state.dropped_since_last_check},
+             {max_length, State#state.max_length},
+             {last_recorded_length, State#state.last_recorded_length},
+             {total_dropped, State#state.total_dropped}],
+    {reply, Stats, State};
+handle_call(sync_check_current_state, From, #state{sync_response_process = undefined} = State) ->
+     self() ! status_ping,
+     {noreply, State#state{sync_response_process = From}};
+handle_call(stop, _From, State) ->
+    lager:info("Stopping Queue Monitor"),
+    {stop,normal,ok,State};
+handle_call(start_timer, _From, #state{timer = undefined} = State) ->
+    TRef = start_update_timer(),
+    {reply, ok, State#state{timer = TRef}};
+handle_call(start_timer, _From, State) ->
+    lager:info("Queue Monitoring timer already started"),
+    {reply, ok, State};
+handle_call(stop_timer, _From, #state{timer = undefined} = State) ->
+    lager:info("Queue Monitoring timer already stopped"),
+    {reply, ok, State};
+handle_call(stop_timer, _From, #state{timer = Timer} = State) ->
+    {ok, cancel} = timer:cancel(Timer),
+    {reply, ok, State#state{timer = undefined}};
+handle_call(Request, _From, State) ->
+    lager:debug("Unknown request: ~p", [Request]),
+    {reply, ignored, State}.
+
+
+%%-------------------------------------------------------------
+%% CASTS
+handle_cast(message_dropped, #state{total_dropped = TotalDropped,
+                                    dropped_since_last_check = Dropped} = State) ->
+    {noreply, State#state{total_dropped = TotalDropped + 1,
+                          dropped_since_last_check = Dropped + 1}};
+handle_cast(Msg, State) ->
+    lager:debug("Unknown cast: ~p", [Msg]),
+    {noreply, State}.
+
+
+%%-------------------------------------------------------------
+%% INFO
+handle_info(reset_dropped_since_last_check, State) ->
+    {noreply,State#state{dropped_since_last_check = 0}};
+handle_info({'EXIT', From, Reason}, #state{worker_process = WorkerPid,
+                                           sync_response_process = SyncPid} = State) ->
+    % Check to see if the EXIT came from our worker
+    case From == WorkerPid of
+        true ->
+              % check if we need to reply to a sync_check_current_state() call
+              case SyncPid of
+                  undefined -> ok;
+                  Pid -> gen_server:reply(Pid, ok)
```

Comment:
`ok` seems like an odd response in this case, should we return an error tuple with the reason?